src/runtime/storage.ts

```diff
@@ -52,18 +74,49 @@ export class Storage {
         }
       });
     }
-    return this.values_[key];
+    return this.values_[finalKey];
+  }
+
+  async set(
+    baseKey: string,
+    value: string,
+    useLocalStorage = false
+  ): Promise<void> {
+    // The old version of storage key without publication identifier.
+    // To be deprecaed in favor of the new version of key.
+    const oldKey = this.getStorageKeyWithoutPublicationId_(baseKey);
+    // The new version of storage key with publication identifier.
+    const newKey = this.getStorageKeyMaybeWithPublicationId_(baseKey);
+    const valueWithNewKey = await this.getInternal_(newKey, useLocalStorage);
+
+    if (
+      valueWithNewKey !== null ||
```

Comment:
Can we add a comment that we explicitly also want to honor the new key when the experiment is off? and same for `remove` function below?